compiler/test/test_end_to_end.re

```diff
@@ -1718,13 +1718,13 @@ let number_tests = [
   te("numbers13", "9 / 0", "denominator of zero"),
   // basic syntax tests
   t("number_syntax1", "1.2", "1.2"),
-  t("number_syntax2", "1.", "1"),
+  t("number_syntax2", "1.", "1.0"),
   t("number_syntax3", ".2", "0.2"),
   te("number_syntax4", ".", "Syntax error"),
   t("number_syntax5", "1.2d", "1.2"),
-  t("number_syntax6", "1.2f", "1.2000000476837158"),
-  t("number_syntax7", "1e2", "100"),
-  t("number_syntax8", "1.2e2", "120"),
+  t("number_syntax6", "1.2f", "1.2000000476837159"),
```

Comment:
It seems you lost some precision. In original implementation it's not happening.
See playground: https://bit.ly/3gKsSsV

expected: `1.2000000476837158`

Reply:
Sorry. It's serialization (not parsing). So originally it also slightly [misprecise](https://www.assemblyscript.org/editor.html#IyFvcHRpbWl6ZT1zcGVlZCZydW50aW1lPW5vbmUKZXhwb3J0IGZ1bmN0aW9uIHRlc3RQYXJzZSgpOiBmMzIgewogIHJldHVybiBGMzIucGFyc2VGbG9hdCgiMS4yIik7Cn0KCmV4cG9ydCBmdW5jdGlvbiB0ZXN0U2VyaWFsaXplKCk6IGJvb2wgewogIHJldHVybiBmMzIoMS4yKS50b1N0cmluZygpID09ICIxLjIwMDAwMDA0NzY4MzcxNTgiOwp9CgpleHBvcnQgZnVuY3Rpb24gdGVzdFNlcmlhbGl6ZTIoKTogYm9vbCB7CiAgcmV0dXJuIGYzMigxLjIpLnRvU3RyaW5nKCkgPT0gIjEuMjAwMDAwMDQ3NjgzNzE1OSI7Cn0KIyFodG1sCjx0ZXh0YXJlYSBpZD0ib3V0cHV0IiBzdHlsZT0iaGVpZ2h0OiAxMDAlOyB3aWR0aDogMTAwJSIgcmVhZG9ubHk+PC90ZXh0YXJlYT4KPHNjcmlwdD4KbG9hZGVyLmluc3RhbnRpYXRlKG1vZHVsZV93YXNtLCB7IC8qIGltcG9ydHMgKi8gfSkKICAudGhlbigoeyBleHBvcnRzIH0pID0+IHsKICAgIGNvbnN0IG91dHB1dCA9IGRvY3VtZW50LmdldEVsZW1lbnRCeUlkKCdvdXRwdXQnKQogICAgb3V0cHV0LnZhbHVlICs9IGBwYXJzZUZsb2F0KDEuMikgPSAke2V4cG9ydHMudGVzdFBhcnNlKCl9XG5gOwogICAgb3V0cHV0LnZhbHVlICs9IGBmMzIoMS4yKS50b1N0cmluZygpID09ICIxLjIwMDAwMDA0NzY4MzcxNTgiIDogJHtleHBvcnRzLnRlc3RTZXJpYWxpemUoKSA/ICJ0cnVlIiA6ICJmYWxzZSJ9XG5gOwogICAgb3V0cHV0LnZhbHVlICs9IGBmMzIoMS4yKS50b1N0cmluZygpID09ICIxLjIwMDAwMDA0NzY4MzcxNTkiIDogJHtleHBvcnRzLnRlc3RTZXJpYWxpemUyKCkgPyAidHJ1ZSIgOiAiZmFsc2UifVxuYDsKICB9KQo8L3NjcmlwdD4=)

Reply:
@MaxGraey Thank you for tracking that down. My understanding, then, is that we inherited this bug/quirk from the AS implementation, rather than introduced it during the integration. Does that sound correct?

Reply:
yes